vfs/fs_utils.cc

```diff
@@ -537,6 +537,33 @@ dirName(const std::string & filename)
 }
 
 /****************************************************************************/
+/* UX FUNCTIONS                                                             */
+/****************************************************************************/
+
+void
+checkWritability(const std::string & url, const std::string & parameterName)
+{
+    // try to create output folder and write open a writer to make sure 
+    // we have permissions before
+    try {
+        Datacratic::makeUriDirectory(url);
```

Comment:
The directory and the file will be created anyways when the procedure runs. Doing it with the filter output streams makes it work for s3 de manière transparente, which is why I did it this way.